lib/rubygems/specification_policy.rb

```diff
@@ -409,6 +413,12 @@ def validate_shebang_line_in(executable)
     warning "#{executable_path} is missing #! line"
   end
 
+  def validate_deprecated_attributes
+    DEPRECATED_ATTRIBUTES.each do |attr|
+      warning("#{attr} is deprecated") unless @specification.send(attr).nil?
```

Comment:
Any idea for better deprecation warning here?

Reply:
Also it would be possible to `error` later in here.

Reply:
Should we follow the same pattern as in `validate_required_attributes`? Defining the list in `Gem::Specification` in that same way would allow us to reuse https://github.com/rubygems/rubygems/pull/3051 and completely get rid of the `rubyforge_project` accessor.

Reply:
Thinking of a good message to give to users, I believe we should probably call this `validate_removed_attributes` since what they have in common is that the fix is to remove the attribute from the gemspec. We could say something like:

```suggestion
      warning("#{attr} is deprecated and ignored. Please remove this attribute from your gemspec to ensure that your gem continues to build in the future.") unless @specification.send(attr).nil?
```

Or a version of that communicating the same message but maybe a bit less threatening :sweat_smile:

Reply:
Sure, that message was just a placeholder for now.

Reply:
Just to make sure @simi, you're planning to continue the work here, right?

Reply:
on it now